test/prettyprinter/Tests/PrettyAstSpec.hs

```diff
@@ -0,0 +1,48 @@
+module Tests.PrettyAstSpec where
+
+import           Test.Hspec                     ( Spec(..)
+                                                , it
+                                                , pending
+                                                , shouldBe
+                                                )
+
+import           Front                          ( renderAst )
+import           Front.Ast
+import           Front.Parser                   ( parseProgram )
+
+parseAndRender :: String -> String
+parseAndRender input = case parseProgram input of
+  Left  e   -> error e
+  Right ast -> renderAst ast
```

Comment:
@j-hui do you think this is a reasonable way to do this? Am I missing some high IQ trick to lift the operations into the `Either` monad or something instead of doing this unwrapping?

Reply:
If you are so inclined, you could write this point-free as:

```haskell
parseAndRender :: String -> String
parseAndRender = bimap error renderAst . parseProgram
```

where `bimap` is a method of the `Bifunctor` type class (you'll need to import it).